code/code/cmd/cmd_bonebreak.cc

```diff
@@ -239,7 +244,15 @@ int bonebreak(TBeing *caster, TBeing *victim)
       SV(SKILL_BONEBREAK);
       rc = bonebreakMiss(caster, victim, 1);
       if (IS_SET_DELETE(rc, DELETE_THIS) || IS_SET_DELETE(rc, DELETE_VICT))
-	return rc;
+        return rc;
+      return TRUE;
+    }
+
+    if (victim->canFocusedAvoidance(bKnown/2)) {
+      SV(SKILL_BONEBREAK);
+      rc = bonebreakMiss(caster, victim, 2);
```

Comment:
Instead of a function call (or perhaps inside that function call), I'd just have an array of string triples. Look up the strings and print them — less code lines and no branch mispredictions in CPU.

Reply:
Not saying you should go and refactor everything in the world, though :)